src/config/migrations/20200616130143_ReplacePermalinkHashIndex.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Yoast SEO Plugin File.
+ *
+ * @package Yoast\WP\SEO\Config\Migrations
+ */
+
+namespace Yoast\WP\SEO\Config\Migrations;
+
+use Yoast\WP\Lib\Migrations\Migration;
+use Yoast\WP\Lib\Model;
+
+/**
+ * ReplacePermalinkHashIndex class.
+ */
+class ReplacePermalinkHashIndex extends Migration {
+
+	/**
+	 * The plugin this migration belongs to.
+	 *
+	 * @var string
+	 */
+	public static $plugin = 'free';
+
+	/**
+	 * Migration up.
+	 *
+	 * @return void
+	 */
+	public function up() {
+		$table_name = $this->get_table_name();
+		$adapter    = $this->get_adapter();
+
+		if ( $adapter->has_table( $table_name ) ) {
```

Comment:
I think the other checks also depend on the table being there.
So perhaps a ! has_table and return instead?
```php
		if ( ! $adapter->has_table( $table_name ) ) {
			return;
```